test/parallel/test-watch-mode-files_watcher.mjs

```diff
@@ -51,6 +51,36 @@ describe('watch mode file watcher', () => {
     assert.strictEqual(changesCount, 1);
   });
 
+  it('should watch changed files with same prefix path string', async () => {
+    const ran = Promise.withResolvers();
+    mkdirSync(tmpdir.resolve('subdir'));
+    mkdirSync(tmpdir.resolve('sub'));
+    const file1 = tmpdir.resolve('subdir', 'file1.mjs');
+    const file2 = tmpdir.resolve('sub', 'file2.mjs');
+    writeFileSync(file2, 'export const hello = () => { return "hello world"; };');
+    writeFileSync(file1, 'import { hello } from "../sub/file2.mjs"; console.log(hello());');
+
+    const child = spawn(process.execPath,
+                        ['--watch', file1],
+                        { encoding: 'utf8', stdio: 'pipe' });
+    let completeCount = 0;
+    child.stdout.on('data', async (data) => {
+      if (data.toString().match(/Completed running/g)) {
+        completeCount++;
+      }
+      if (completeCount === 1) {
+        ran.resolve();
+      }
+      // The file is reloaded due to file watching
+      if (completeCount === 2) {
+        child.kill();
+        await once(child, 'exit');
+      }
+    });
+    await ran.promise;
+    appendFileSync(file1, '\n // append 1');
```

Comment:
Using an async event handler is bad practice, we should either await the promise outside of the event handler, or not await it at all
```suggestion
    const exited = Promise.withResolvers();
    child.stdout.on('data', (data) => {
      if (data.toString().match(/Completed running/g)) {
        completeCount++;
      }
      if (completeCount === 1) {
        ran.resolve();
      }
      // The file is reloaded due to file watching
      if (completeCount === 2) {
        child.once('exit', exited.resolve);
        child.kill();
      }
    });
    await ran.promise;
    appendFileSync(file1, '\n // append 1');
    await exited.promise;
```